src/Parsers/ASTDatabaseOrNone.cpp

```diff
@@ -0,0 +1,16 @@
+#include <Parsers/ASTDatabaseOrNone.h>
+#include <IO/Operators.h>
+
+namespace DB
+{
+void ASTDatabaseOrNone::formatImpl(const FormatSettings & settings, FormatState &, FormatStateStacked) const
+{
+    if (none)
+    {
+        settings.ostr << (settings.hilite ? IAST::hilite_keyword : "") << "NONE" << (settings.hilite ? IAST::hilite_none : "");
+        return;
+    }
+    settings.ostr << database_name;
```

Comment:
`database_name` can contain spaces, so it's better to use `backQuoteIfNeed()` here to enclose the database name in backticks in this case:

```
settings.ostr << backQuoteIfNeed(database_name);
```

Reply:
https://github.com/ClickHouse/ClickHouse/pull/27702